### ISSUE TITLE
A Systematic Evaluation of The Trained ML Model

### ISSUE THREAD

**Issue by hehao98** [osslab-pku/gfi-bot#40]
To align with the requirements of a research paper, the following evaluations need to be conducted:

1. Comparison with alternative models (LightGBM, Logistic Regression, SVM, DNN, and Random Forest)
2. Ablation studies on features (along multiple dimensions of features)
3. Feature importance analysis
4. Showing the advantage of automated hyperparameter tuning

All evaluation data should be stored in the MongoDB database (for visualization in the frontend in the future).

**Comment by minghuizhou** [comment]
Could you assign the tasks to people?  :-)

**Comment by hehao98** [comment]
> Could you assign the tasks to people? :-)

Runzhi has invited Haonan for some of the frontend tasks, but we are not sure which one he will work on first, so these tasks are still unassigned.

Runzhi also told me that the frontend code base is a little bit difficult to understand, maybe we need to invest some time to digest the code itself before we can assign frontend tasks to others  ;<

**Comment by hehao98** [comment]
[GFI-Bot] Predicted as Good First Issue with probability 1.09%.

**Comment by arsyadal** [comment]
Hi @hehao98 @minghuizhou 👋, I have implemented the systematic ML evaluation suite (model comparison across RandomForest, GradientBoosting, LogisticRegression, SVM, MLP, ablation studies, feature importance extraction, MongoDB \ModelEvaluation\ collection, and REST API \/api/model/evaluation\). The implementation is submitted in PR #69.
